tests/accelerators/test_hpu.py

```diff
@@ -0,0 +1,358 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from argparse import ArgumentParser
+from typing import Optional
+
+import pytest
+import torch
+import torch.nn.functional as F
+
+from pytorch_lightning import Callback, seed_everything, Trainer
+from pytorch_lightning.accelerators import CPUAccelerator, HPUAccelerator
+from pytorch_lightning.callbacks import HPUStatsMonitor
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.plugins import HPUPrecisionPlugin
+from pytorch_lightning.strategies.hpu import HPUStrategy
+from pytorch_lightning.trainer.states import RunningStage, TrainerFn
+from pytorch_lightning.trainer.supporters import CombinedLoader
+from pytorch_lightning.utilities import _AcceleratorType, _HPU_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.datamodules import ClassifDataModule
+from tests.helpers.runif import RunIf
+from tests.helpers.simple_models import ClassificationModel
+
+if _HPU_AVAILABLE:
+    import habana_frameworks.torch.core as htcore
+
+    os.environ["PL_TORCH_DISTRIBUTED_BACKEND"] = "hccl"
+
+
+class HPUModel(BoringModel):
```

Comment:
Why is there a need to subclass BoringModel here?

Reply:
This is just a sample test

Reply:
@jerome-habana I don't see this class specific to HPU tests, so shall we use `BoringModel` directly without subclassing it?

Reply:
I just wanted to have one class for HPU here so that it aids for adding additional tests with callbacks as we progress.

Reply:
I see your points, but I still believe customized BoringModel should only be used for test cases where its customization is needed. So, for now, I think we should use BoringModel here.

Also commented at https://github.com/PyTorchLightning/pytorch-lightning/pull/10404#discussion_r744565080.

Reply:
Is it mandatory for tests ?